job_creator.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+class JobCreator
+{
+    private $installerVersion = null;
+
+    /**
+     * Get the correct version of silverstripe/installer to include for the given repository and branch
+     */
+    public function getInstallerVersion(string $githubRepository, string $branch): string
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        // e.g. ['4', '11']
+        $portions = explode('.', $branch);
+        if (count($portions) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $portions[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            // remove '4' version
+            $installerVersions = array_diff($installerVersions, ['4']);
+            // get the minor portions of the verisons e.g. [9, 10, 11]
+            $minorPortions = array_map(fn($portions) => (int) explode('.', $portions)[1], $installerVersions);
+            sort($minorPortions);
+            return $minorPortions[count($minorPortions) - 1] . '.x-dev';
+        }
+    }
+    
+    public function createJob(int $phpIndex, array $opts): array
+    {
+        $installerKey = str_replace('.x-dev', '', $this->installerVersion);
+        $installerKey = $installerKey ?: '4';
+        $phpVersions = INSTALLER_TO_PHP_VERSIONS[$installerKey];
+        $default = [
+            # ensure there's a default value for all possible return keys
+            # this allows us to use `if [[ "${{ matrix.key }}" == "true" ]]; then` in github-actions-ci-cd/ci.yml
+            'installer_version' => $this->installerVersion,
+            'php' => $phpVersions[$phpIndex] ?? $phpVersions[count($phpVersions) - 1],
+            'db' => DB_MYSQL_57,
+            'composer_require_extra' => '',
+            'composer_args' => '',
+            'name_suffix' => '',
+            'phpunit' => false,
+            'phpunit_suite' => 'all',
+            'phplinting' => false,
+            'phpcoverage' => false,
+            'endtoend' => false,
+            'endtoend_suite' => 'root',
+            'endtoend_config' => '',
```

Comment:
Is this ever anything other than a blank string? I can't find anywhere else this gets set.

Reply:
It's settable via extra_jobs